client/gc_test.go

```diff
@@ -0,0 +1,325 @@
+	if len(m.availableValues) == 0 {
+		return nil
+	}
+
+	available := m.availableValues[m.index]
+	usedPercent := m.usedPercents[m.index]
+	inodePercent := m.inodePercents[m.index]
+
+	if m.index < len(m.availableValues)-1 {
+		m.index = m.index + 1
+	}
+	return &stats.HostStats{
+		AllocDirStats: &stats.DiskStats{
+			Available:         available,
+			UsedPercent:       usedPercent,
+			InodesUsedPercent: inodePercent,
+		},
+	}
+}
+
+func TestAllocGarbageCollector_MarkForCollection(t *testing.T) {
+	logger := log.New(os.Stdout, "", 0)
+	gc := NewAllocGarbageCollector(logger, &MockStatsCollector{}, 0)
+
+	_, ar1 := testAllocRunnerFromAlloc(mock.Alloc(), false)
+	if err := gc.MarkForCollection(ar1); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	gcAlloc := gc.allocRunners.Pop()
+	if gcAlloc == nil || gcAlloc.allocRunner != ar1 {
+		t.Fatalf("bad gcAlloc: %v", gcAlloc)
+	}
+}
+
+func TestAllocGarbageCollector_Collect(t *testing.T) {
+	logger := log.New(os.Stdout, "", 0)
+	gc := NewAllocGarbageCollector(logger, &MockStatsCollector{}, 0)
+
+	_, ar1 := testAllocRunnerFromAlloc(mock.Alloc(), false)
+	_, ar2 := testAllocRunnerFromAlloc(mock.Alloc(), false)
+	if err := gc.MarkForCollection(ar1); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if err := gc.MarkForCollection(ar2); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if err := gc.Collect(ar1.Alloc().ID); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	gcAlloc := gc.allocRunners.Pop()
+	if gcAlloc == nil || gcAlloc.allocRunner != ar2 {
+		t.Fatalf("bad gcAlloc: %v", gcAlloc)
+	}
+}
+
+func TestAllocGarbageCollector_CollectAll(t *testing.T) {
+	logger := log.New(os.Stdout, "", 0)
+	gc := NewAllocGarbageCollector(logger, &MockStatsCollector{}, 0)
+
+	_, ar1 := testAllocRunnerFromAlloc(mock.Alloc(), false)
+	_, ar2 := testAllocRunnerFromAlloc(mock.Alloc(), false)
+	if err := gc.MarkForCollection(ar1); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if err := gc.MarkForCollection(ar2); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if err := gc.CollectAll(); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	gcAlloc := gc.allocRunners.Pop()
+	if gcAlloc != nil {
+		t.Fatalf("bad gcAlloc: %v", gcAlloc)
+	}
+}
+
+func TestAllocGarbageCollector_MakeRoomForAllocations_EnoughSpace(t *testing.T) {
+	logger := log.New(os.Stdout, "", 0)
+	statsCollector := &MockStatsCollector{}
+	gc := NewAllocGarbageCollector(logger, statsCollector, 20)
+
+	_, ar1 := testAllocRunnerFromAlloc(mock.Alloc(), false)
+	close(ar1.waitCh)
+	_, ar2 := testAllocRunnerFromAlloc(mock.Alloc(), false)
+	close(ar2.waitCh)
+	if err := gc.MarkForCollection(ar1); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if err := gc.MarkForCollection(ar2); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// Make stats collector report 200MB free out of which 20MB is reserved
+	statsCollector.availableValues = []uint64{200 * MB}
+	statsCollector.usedPercents = []float64{0}
+	statsCollector.inodePercents = []float64{0}
+
+	alloc := mock.Alloc()
```

Comment:
Can you set the disk un the tests so it is obvious in the test and the tests become resilient to changes to the mock alloc